### PR TITLE
Enhance plugin-descriptor.properties guide

### PR DESCRIPTION
**Pull request by dadoonet**
- extract doc from the descriptor
- make obvious that plugin authors will have to release a new version for each elasticsearch version

### CODE REVIEW

**Comment by dadoonet** [comment]
@rmuir or @clintongormley do you agree with this doc?

File: dev-tools/src/main/resources/plugin-metadata/plugin-descriptor.properties [hunk 1]
```diff
@@ -64,6 +64,9 @@ classname=${elasticsearch.plugin.classname}
 java.version=${maven.compiler.target}
 #
 # 'elasticsearch.version' version of elasticsearch compiled against
+# You will have to release a new version of the plugin for each new
+# elasticsearch release. This version is checked when the plugin
+# is loaded so non-compatible plugins are not loaded.
```
Comment (clintongormley) [inline dev-tools/src/main/resources/plugin-metadata/plugin-descriptor.properties]: Actually, Elasticsearch will refuse to start in the presence of plugins with the incorrect `elasticsearch.version`.

**Comment by clintongormley** [comment]
Added one comment.  Also, in https://github.com/dadoonet/elasticsearch/blob/doc/add-elasticsearch-version/dev-tools/src/main/resources/plugin-metadata/plugin-descriptor.properties#L28 the example uses an elasticsearch version of "2.0", which isn't correct. It should use a full version number.  Could you update that as well please?

**Comment by dadoonet** [comment]
Thanks @clintongormley for the review.
Merged in 2.0 to master branch.

**Pull request merged**